rust-plugins/sass/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+#![deny(clippy::all)]
+
+use farmfe_core::{config::Config, module::ModuleType, plugin::Plugin};
+use farmfe_macro_plugin::farm_plugin;
+use farmfe_toolkit::{fs, regex::Regex};
+use sass_embedded::{Sass, StringOptions};
+use std::env;
+use std::env::consts::{ARCH, OS};
+use std::path::PathBuf;
+
+const PKG_NAME: &str = "@farmfe/plugin-sass";
+
+fn get_os() -> &'static str {
+  match OS {
+    "linux" => "linux",
+    "macos" => "darwin",
+    "windows" => "win32",
+    os => panic!("dart-sass-embed is not supported OS: {}", os),
+  }
+}
+
+fn get_arch() -> &'static str {
+  match ARCH {
+    "x86" => "ia32",
+    "x86_64" => "x64",
+    "aarch64" => "arm64",
+    arch => panic!("dart-sass-embed is not supported arch: {}", arch),
+  }
+}
+
+fn get_exe_path() -> PathBuf {
+  let os = get_os();
+  let arch = get_arch();
+  let entry_file = if let "win32" = os {
+    "dart-sass-embedded.bat"
+  } else {
+    "dart-sass-embedded"
+  };
+  let cur_dir = env::current_dir().unwrap().to_string_lossy().to_string();
+  // user manually installs related dependencies
+  let manual_installation_path = PathBuf::from(&cur_dir)
+    .join(format!("node_modules/sass-embedded-{os}-{arch}"))
```

Comment:
Two quetions:
1. Does user need to install sass-embedded into their project
2. Does it work with pnpm